src/compiler/types.ts

```diff
@@ -1473,7 +1473,8 @@ module ts {
         EnumMemberExcludes = Value,
         FunctionExcludes = Value & ~(Function | ValueModule),
         ClassExcludes = (Value | Type) & ~ValueModule,
-        InterfaceExcludes = Type & ~Interface,
+        AmbientClassExcludes = (Value | Type) & ~(ValueModule | Interface),

```

Comment:
why not just letting classes and interfaces merge, and reporting an error later on in CheckClassDeclaration for instance if the class is not ambient? just like we do with merging order for module/class and module/class being in different files.